scripts/lib.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+check_program(){
+	type "$1" >/dev/null 2>&1
+}
+
+check_root()
+{
+	if [ "$(id -u)" != "0" ]; then
+		echo "Root is needed"
+		exit 1
+	fi
+}
+
+generate_dnf_config()
+{
+	REPO_NAME=${REPO_NAME:-"base"}
+	CACHE_DIR=${CACHE_DIR:-"/var/cache/dnf-${OS_NAME}"}
+	cat > "${DNF_CONF}" << EOF
+[main]
+cachedir=${CACHE_DIR}
+keepcache=0
+debuglevel=2
+logfile=/var/log/dnf.log
+exactarch=1
+obsoletes=1
+gpgcheck=0
+plugins=0
+installonly_limit=3
+#Dont use the default dnf reposdir
+#this will prevent to use host repositories
+reposdir=/root/mash
+retries=5
+
+[${REPO_NAME}]
+name=${OS_NAME}-${OS_VERSION} ${REPO_NAME}
+failovermethod=priority
+metalink=${REPO_URL}
+enabled=1
+EOF
+	# Handle OS-specific extra repos
+	# Add extra repos for CentOS
+	if [ "$OS_NAME" == "Centos" ];then
```

Comment:
Instead of handle an special case for each distro, Can we just have a function `create_dnf_config` what uses OS_NAME, CACHE_DIR,  LOG_FILE, GPG_KEY, ADDITIONAL_REPOS_LIST.
And use the rest of of default options.